src/pages/cms-page/[...slug].tsx

```diff
@@ -54,13 +54,62 @@ export async function getStaticPaths() {
   return { paths: [], fallback: false };
 }
 
-export async function getStaticProps(context: GetStaticPropsContext): Promise<
-  GetStaticPropsResult<{
-    initialApolloState: NormalizedCacheObject;
-    page: PageFieldsFragment;
-    breadcrumbs: Breadcrumb[];
-  }>
-> {
+type ResultProps =
+  | {
+      initialApolloState: NormalizedCacheObject;
+      page: PageFieldsFragment;
+      breadcrumbs: Breadcrumb[];
+    }
+  | {
+      error?: {
+        statusCode: number;
+      };
+    };
+
+export async function getStaticProps(
+  context: GetStaticPropsContext
+): Promise<GetStaticPropsResult<ResultProps>> {
+  try {
+    const {
+      currentPage: page,
+      breadcrumbs,
+      cmsClient,
+    } = await getProps(context);
+
+    if (!page) {
+      return {
+        notFound: true,
+        revalidate: true,
+      };
+    }
+
+    return {
+      props: {
+        initialApolloState: cmsClient.cache.extract(),
+        ...(await serverSideTranslations(
+          context.locale as string,
+          ALL_I18N_NAMESPACES
+        )),
+        page,
+        breadcrumbs,
+      },
+      revalidate: 60,
+    };
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.log('Error while generating content page', e);
+    return {
+      props: {
+        error: {
+          statusCode: 400,
+        },
+      },
+      revalidate: 10,
+    };
+  }
+}
```

Comment:
I always wondered whether there was a way to declare this logic so that it could be composed and pages could share it. For example:

```js
function computeStaticProps(executeFunction, config) {
  try {
    executeFunction()
  } catch (e) {
    console.error()
    return {} // config
  }
}
```

I wasn't confident enough in that it made sense so I just inlined everything and committed a bit of copy pasting.

In this application it's really not a necessity considering you are only declaring this logic once. But I'm still wondering whether you had any similar thoughts?

Reply:
I didn't exactly have those thoughts because just like you said in this application there are no similar `getStaticProps` implementations. My main reason was to avoid too long try catch section and also to make it little bit easier to understand.

I think sharing logic might make sense in liikunta.